npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npmrc_builder.rb

```diff
@@ -155,6 +157,12 @@ def dependency_urls
                           .flatten
                           .select { |url| url.is_a?(String) }
                           .reject { |url| url.start_with?("git") }
+          elsif shrinkwrap
+            @dependency_urls +=
+              shrinkwrap.content.scan(/"resolved"\s*:\s*"(.*)"/)
```

Comment:
seems to be the same code pattern as the preceding block; an opportunity to consolidate?